src/grpc.ts

```diff
@@ -347,26 +378,38 @@ export class GrpcClient {
    * @param {number} options.port - The port of the service.
    * @param {grpcTypes.ClientCredentials=} options.sslCreds - The credentials to be used
    *   to set up gRPC connection.
+   * @param {string} defaultServicePath - The default service path.
    * @return {Promise} A promise which resolves to a gRPC stub instance.
    */
-  async createStub(CreateStub: typeof ClientStub, options: ClientStubOptions) {
+  async createStub(
+    CreateStub: typeof ClientStub,
+    options: ClientStubOptions,
+    customServicePath?: boolean
+  ) {
     // The following options are understood by grpc-gcp and need a special treatment
     // (should be passed without a `grpc.` prefix)
     const grpcGcpOptions = [
       'grpc.callInvocationTransformer',
       'grpc.channelFactoryOverride',
       'grpc.gcpApiConfig',
     ];
-    const serviceAddress = options.servicePath + ':' + options.port;
-    const creds = await this._getCredentials(options);
+    const [cert, key] = await this._detectClientCertificate(options);
```

Comment:
Is this credentials workflow applying to the REST transport?

I am thinking we should move the functions out of grpc class. It is more like utility functions and potentially reused in REST transport. @alexander-fenster what do you think?

To unblock @bcoe, we could merge this PR and refactor moving to `credentials.ts`in a seperate PR.

Reply:
@summer-ji-eng most pressing today is the gRPC transport, mTLS is supported by the following APIs:

Compute, Pubsub, KMS, Logging, Spanner, and BigTable, BigQuery, Storage.

BigQuery and Storage do not use gax-nodejs, and are will be able to support mTLS through [this change](https://github.com/googleapis/gaxios/pull/399) in gaxios.

KMS, Logging, Spanner, and BigTable are primarily gRPC, and will use mTLS through the gRPC transport.

---

We should think about supporting mTLS in the REST transport for Compute ... but, we should be careful because I believe the other reason folks use the REST transport is for Webpacking client-side versions of our libraries.

The implementation of mTLS will not work well when Webpacked.

Reply:
Thanks for explaining. Considering the YAGNI smell, I agree that leave the code in `grpc.ts`.